lib/rules/no-test-id-queries.ts

```diff
@@ -0,0 +1,47 @@
+import { TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import { ALL_QUERIES_VARIANTS } from '../utils';
+
+export const RULE_NAME = 'no-test-id-queries';
+export type MessageIds = 'noTestIdQueries';
+type Options = [];
+
+const QUERIES_REGEX = `/^(${ALL_QUERIES_VARIANTS.join('|')})TestId$/`;
+
+export default createTestingLibraryRule<Options, MessageIds>({
+	name: RULE_NAME,
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Ensure no `data-testid` queries are used',
+			recommendedConfig: {
+				dom: 'error',
+				angular: 'error',
+				react: 'error',
+				vue: 'error',
+				svelte: 'error',
+				marko: 'error',
+			},
+		},
```

Comment:
suggestion (blocking): Please, remove this rule from all the recommended configs.

Based on [the package Semantic Versioning Policy](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/semantic-versioning-policy.md), adding more rules to existing shareable configurations is a breaking change. So we will release the new rule as a minor, then in the next major will be added to all configs.

Reply:
oops, sorry, it was copy-pasted

Reply:
No worries, it happens all the time! Would be a good idea to set up an internal tool for creating new rules.